src/pyFAI/gui/pilx/MainWindow.py

```diff
@@ -267,44 +290,74 @@ def onMouseClickOnImage(self, x: float, y: float):
             radial_value + self._delta_radial_over_2,
         )
 
-    def getColorLastCurve(self):
-        curves = self._integrated_plot_widget.getAllCurves()
-        if curves:
-            return self._integrated_plot_widget.getAllCurves()[-1].getColor()
+    def getColorCurve(self, legend: str):
+        curve = self._integrated_plot_widget.getCurve(legend=legend)
+        if curve:
+            return curve.getColor()
         else:
             return
 
+    def getColorAvailable(self, legend: str):
+        if self._integrated_plot_widget.getCurve(legend=f"INTEGRATE{legend}"):
+            color = self._integrated_plot_widget.getCurve(legend=f"INTEGRATE{legend}").getColor()
+        else:
+            index_color = 0
+            color = rgba(self._integrated_plot_widget.getDefaultColors()[index_color])
+            while color in [self._integrated_plot_widget.getCurve(legend=legend).getColor() for legend in self._integrated_plot_widget]:
+                index_color += 1
+                color = rgba(self._integrated_plot_widget.getDefaultColors()[index_color])
+        return color
+
     def setNewBackgroundCurve(self, x: float, y: float):
         indices = self._map_plot_widget.getImageIndices(x, y)
         if indices is None:
             return
         radial_background, pattern_background = self.getBackgroundCurve(row=indices.row, col=indices.col)
 
-        # Unset the background if it's the same pixel
-        if self._intensity_background_curve:
-            if (indices.row, indices.col) in self._intensity_background_curve.keys():
-                self._intensity_background_curve = {}
-                markers = [item for item in self._map_plot_widget.getItems() if isinstance(item, MarkerBase)]
-                _ = [self._map_plot_widget.removeMarker(legend=marker.getLegend()) for marker in markers if "BG" in marker.getLegend()]
-
-                for legend, intensity_curve in self._intensity_curves_cache.items():
-                    self._integrated_plot_widget.addCurve(
-                        x=radial_background,
-                        y=intensity_curve,
-                        legend=legend,
-                        selectable=False,
-                        color=self._integrated_plot_widget.getCurve(legend=legend).getColor()
-                    )
-                return
+        # Unset the background if it's the same pixel and delete markers
+        if (indices.row, indices.col) in self._intensity_background_curve.keys():
+            self._intensity_background_curve = {}
+            markers = [item for item in self._map_plot_widget.getItems() if isinstance(item, MarkerBase)]
+            _ = [self._map_plot_widget.removeMarker(legend=marker.getLegend()) for marker in markers if "BG" in marker.getLegend()]
+        else:
```

Comment:
Since you don't need to execute the following instructions if the background is unset, you can `return` after removing the corresponding marker.

This `else` and the `if self._intensity_background_curve:` below then become unnecessary

Reply:
It has to be executed, to plot the curves again without any subtraction